src/matlab2tikz.m

```diff
@@ -3143,10 +3143,19 @@ case size(cData,2) % specific y-data is given
     prop         = switchMatOct(m2t, 'BarPeers', 'bargroup');
     bargroup     = get(h, prop);
     numBarSeries = numel(bargroup);
+    
+    % HG2 handling:
+    % - BarPeers are sorted in reversed order wrt HG1
     if isHG2(m2t)
         bargroup = bargroup(end:-1:1);
     end
-    [~, barSeriesId] = ismember(handle(h), bargroup);
+    % - Ensure graphic classes are double handles 
+    %   NOTE: Octave does not support handle()
+    h        = double(h);
```

Comment:
I thought I saw somewhere that converting the new HG2 objects to doubles would be deprecated in the future, however, I cannot find a source for that claim.

But it makes me wonder: maybe we should just make the `handle()` call dependent on Octave/ML instead of using a cast to doubles, e.g. `if isMATLAB, h = handle(h), end`.
Also, I have to say I'm unsure why we would need to call the `handle` function. So probably that needs to be documented (or even removed, and then the problem would be solved as well).